pkg/driver/utils.go

```diff
@@ -106,3 +107,29 @@ func getCapacity(cvc *apisv1.CStorVolumeConfig) string {
 	qCap := cvc.Spec.Capacity[corev1.ResourceStorage]
 	return qCap.String()
 }
+
+func getVolumeCondition(vol *apisv1.CStorVolume) *csi.VolumeCondition {
+	condition := &csi.VolumeCondition{}
+	if vol.Status.Phase != apisv1.CVStatusHealthy {
+		condition.Abnormal = true
+	}
+
+	switch vol.Status.Phase {
+	case apisv1.CVStatusHealthy:
+		condition.Message = "Volume status is up"
+
+	case apisv1.CVStatusInit:
+		condition.Message = "Volume status is in init state"
+
+	case apisv1.CVStatusOffline:
+		condition.Message = "Volume status is offline"
+
+	case apisv1.CVStatusDegraded:
+		condition.Message = "Volume status is degraded"
+
+	default:
+		condition.Message = "Volume status is unknown"
+	}
```

Comment:
Since [message](https://github.com/kubernetes-csi/external-health-monitor/blob/0e1ab70ac6b33a2f05e6e057039c709c19b9ffe0/pkg/csi-handler/pv_checker.go#L169-L172) is propagated as an event let's put more info for being in these states.  For example: 
```go
case apisv1.CVStatusInit:
       condition.Message = fmt.Sprintf("Volume is getting initialized, replicas are not yet connected to the target Healty replicas: %d, Degraded replicas: %d, Offline replicas: %d Total replicas: %d", getHealthyReplicaCount(vol), getDegradedReplicaCount(vol), getOfflineReplicaCount(vol), vol.Spec.ReplicationFactor)
       ...
       ...
       ...
```
With above user no need to look at other resources(WDYT).

Reply:
Also once we fix the CV status based on the target pod availablility, we can rely on that info and can use that further.